File: christmasTree.py
n = 5
pattern = ''
index = 0
#first |
for first in range(0,n+1):
    #add n times blank space
    pattern = pattern + ' '
#in the middle is the |
pattern = pattern + '|'
#rest of the symbols
for i in range(1,n):
    #first add a new line
    pattern = pattern + '\n'
    #we need a variable (index) to count how many
    #times we add blank space (for each line the number is 
    #going to be smaller, because of the more stars)
    index+=1

    for space in range(index,n+1):
        #add blank space
        pattern = pattern + ' '
    
    for a in range(0,i):
        #this one add stars BEFORE the line in the middle
        #a is compared to i, because i doesn't reset
        pattern = pattern + '*'
    #add the line in the middle
    pattern = pattern + '|'

    for b in range(0,i):
        #this one is for AFTER the line
        pattern = pattern + '*'
#print
print(pattern)

File: christmasTree.js
let n = 5;
let pattern = '';
let index = 0;
//first |
for(first = 0;first<=n;first++){
    //add n times blank space
    pattern = pattern + ' '
}
//in middle is the |
pattern = pattern + '|'
//rest of symbols
for(i=1;i<n;i++){
    //first add a new line
    pattern = pattern + '\n';
    /*we need a variable (index) to count how many
    times we add blank space (for each line the number is 
    going to be smaller, because of the more stars)*/
    index++;

    for(space=index;space<=n;space++){
        //add blank space
        pattern = pattern + ' ';
    }

    for(a=0;a<i;a++){
        //this one adds stars BEFORE the line |
        //a is compared to i, because i doen't reset
        pattern = pattern + '*';
        
    }
    //add line in the middle
    pattern = pattern + '|';
    for(b=0;b<i;b++){
        //this one is for AFTER the line
        pattern = pattern + '*';
    }
}
//print
console.log(pattern)